mmv1/products/discoveryengine/SearchEngine.yaml

```diff
@@ -0,0 +1,147 @@
+    primary_resource_id: 'basic'
+    primary_resource_name:
+      'fmt.Sprintf("tf_test_search_engine%s", context["random_suffix"])'
+    vars:
+      engine_id: "example-engine-id"
+      data_store_id: "example-datastore-id"
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'engine_id'
+    description: |
+      Unique ID to use for Search Engine App.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'collection_id'
+    description: |
+      The collection ID.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    description: |
+      Location.
+    required: true
+    immutable: true
+    url_param_only: true
+
+properties:
+  - !ruby/object:Api::Type::Enum
+    name: 'industryVertical'
+    description: |
+      Industry Vertical
+    immutable: true
+    values:
+      - :GENERIC
+      - :MEDIA
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      Display Name
+    required: true
+  - !ruby/object:Api::Type::Array
+    name: 'dataStoreIds'
+    description: |
+      Data Store ID
+    required: true
+    immutable: true
+    item_type: Api::Type::String
+  - !ruby/object:Api::Type::Enum
+    name: 'solutionType'
+    description: |
+      A solution type
+    immutable: true
+    values:
+      - :SOLUTION_TYPE_SEARCH
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: |
+      Timestamp the Engine was created at.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: |
+      Timestamp the Engine was last updated.
+    output: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'searchEngineConfig'
+    description: |
+      Configurations for a Search Engine.
+    required: false
+    immutable: false
+    properties:
+      - !ruby/object:Api::Type::Enum
+        name: 'searchTier'
+        description: |
+          Search Tier
+        required: false
+        immutable: false
+        values:
+          - :SEARCH_TIER_STANDARD
+          - :SEARCH_TIER_ENTERPRISE
+      - !ruby/object:Api::Type::Array
+        name: 'searchAddOns'
+        description: |
+          The add-on that this search engine enables.
+        required: false
+        immutable: false
+        item_type: !ruby/object:Api::Type::Enum
+          name: 'searchAddOn'
+          description: |
+            The add-on that this search engine enables.
+          values:
+            - :SEARCH_ADD_ON_LLM
+  - !ruby/object:Api::Type::NestedObject
+    name: 'commonConfig'
+    description: |
+      Configurations for a Search Engine.
+    required: false
+    immutable: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'companyName'
+        description: |
+          The name of the company, business or entity that is associated with the engine
+        required: false
```

Comment:
let's make the `companyName` required since this is the only field in the block. We can change it to optional if more fields are added in this block.

Reply:
@chenlei1216

Reply:
Doesn't hurt to accept an empty block from our backend. I'm fine with the current setting that we don't have to change this in the future (which we may forget).